### PR TITLE
[SofaCommon] Pluginize SofaObjectInteraction

### PR DESCRIPTION
**Pull request by fredroy**
#1597

- Usual cleaning (pragma, namespaces)
- warnings removal
- index/size uniformization

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaObjectInteraction/src/SofaObjectInteraction/initSofaObjectInteraction.h [hunk 1]
```diff
@@ -19,25 +19,10 @@
...
-#include <SofaObjectInteraction/initObjectInteraction.h>
+#pragma once
+#include <SofaObjectInteraction/config.h>
 
-
-namespace sofa
-{
-
-namespace component
+namespace sofa::component
```
Comment (jnbrunet) [inline SofaKernel/modules/SofaObjectInteraction/src/SofaObjectInteraction/initSofaObjectInteraction.h:25]: Why keeping this file at this point?

Comment (fredroy) [reply SofaKernel/modules/SofaObjectInteraction/src/SofaObjectInteraction/initSofaObjectInteraction.h:25]: at the beginning, just to get an header next to initBlaBla.cpp but yeah not useful, I concur

Comment (fredroy) [reply SofaKernel/modules/SofaObjectInteraction/src/SofaObjectInteraction/initSofaObjectInteraction.h:25]: for SofaBase modules, it is useful tho

Comment (jnbrunet) [reply SofaKernel/modules/SofaObjectInteraction/src/SofaObjectInteraction/initSofaObjectInteraction.h:25]: Yeah but for plugins, the initBlabla is only there to expose global functions to get informations about the plugin by the plugin manager, no need for a header file.

Comment (jnbrunet) [reply SofaKernel/modules/SofaObjectInteraction/src/SofaObjectInteraction/initSofaObjectInteraction.h:25]: Anyway, keep it or delete it, I don't really care, everything looks good to me, ready to be merged whenever you are ready 👍

**Pull request merged**